content/en/docs/languages/net/instrumentation.md

```diff
@@ -63,320 +281,388 @@ using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+have an initialized
+[`TracerProvider`](/docs/concepts/signals/traces/#tracer-provider) that will let
+you create a [`Tracer`](/docs/concepts/signals/traces/#tracer).
 
-// Configure important OpenTelemetry settings, the console exporter
-builder.Services.AddOpenTelemetry()
-  .WithTracing(b =>
-  {
-      b
-      .AddSource(serviceName)
-      .ConfigureResource(resource =>
-          resource.AddService(
-            serviceName: serviceName,
-            serviceVersion: serviceVersion))
-      .AddAspNetCoreInstrumentation()
-      .AddConsoleExporter();
-  });
-```
+If a `TracerProvider` is not created, the OpenTelemetry APIs for tracing will
+use a no-op implementation and fail to generate data.
 
-This is also where you can configure instrumentation libraries.
+If you followed the instructions to [initialize the SDK](#initialize-the-sdk)
+above, you have a `TracerProvider` setup for you already. You can continue with
+[acquiring a tracer](#acquiring-a-tracer).
 
-Note that this sample uses the Console Exporter. If you are exporting to another
-endpoint, you'll have to use a different exporter.
 
 ### Setting up an ActivitySource
 
-Once tracing is initialized, you can configure an
+Anywhere in your application where you write manual tracing code should configure an
 [`ActivitySource`](/docs/concepts/signals/traces/#tracer), which will be how you
 trace operations with [`Activity`](/docs/concepts/signals/traces/#spans)
 elements.
 
-Typically, an `ActivitySource` is instantiated once per app/service that is
-being instrumented, so it's a good idea to instantiate it once in a shared
-location. It is also typically named the same as the Service Name.
+For example:
 
 ```csharp
-using System.Diagnostics;
 
 public static class Telemetry
 {
     //...
 
-    // Name it after the service name for your app.
-    // It can come from a config file, constants file, etc.
-    public static readonly ActivitySource MyActivitySource = new(TelemetryConstants.ServiceName);
+    public static readonly ActivitySource MyActivitySource = new("name", "version");
 
     //...
 }
 ```
+The values of name and version should uniquely identify the Instrumentation Scope, such as the package, module or class name. While the name is required, the version is still recommended despite being optional.
 
-You can instantiate several `ActivitySource`s if that suits your scenario,
-although it is generally sufficient to just have one defined per service.
-
-### Creating Activities
+It’s generally recommended to define `ActivitySource` once per app/service that is been instrumented, but you can instantiate several `ActivitySource`s if that suits your scenario
 
-To create an [`Activity`](/docs/concepts/signals/traces/#spans), give it a name
-and create it from your
-[`ActivitySource`](/docs/concepts/signals/traces/#tracer).
+In the case of the example app, we will create a new file `Instrumentation.cs` as a custom type to hold reference for the ActivitySource. 
```

Comment:
it looks like there is already a `public static class Telemetry` shown above. That must be removed if going with `instrumentation.cs`?